src/DynamoUtilities/Md2Html.cs

```diff
@@ -93,53 +96,63 @@ protected static string GetToolPath(string relativePath)
             var toolPath = Path.Combine(rootPath,relativePath);
             return toolPath;
         }
-        //TODO if we see any issues with deadlocks we can try using a timeout on another thread.
         /// <summary>
         /// Read data from CLI tool
-        /// <returns>Returns data read from CLI tool</returns>
         /// </summary>
-        protected virtual string GetData()
+        /// <param name="timeoutms">will return empty string if we don't finish reading all data in the timeout provided in milliseconds.</param>
+        /// <returns></returns>
+        protected virtual async Task<string> GetData(int timeoutms)
         {
-            if (process.HasExited)
+            var readStdOutTask = Task.Factory.StartNew(() =>
```

Comment:
Why not [Task.Run](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.run?view=net-8.0) ?